engine/consensus/compliance/engine_test.go

```diff
@@ -210,3 +211,23 @@ func (cs *ComplianceSuite) TestProcessUnsupportedMessageType() {
 	require.Error(cs.T(), err)
 	require.True(cs.T(), engine.IsIncompatibleInputTypeError(err))
 }
+
+// TestOnFinalizedBlock tests if finalized block gets processed when send through `Engine`.
+// Tests the whole processing pipeline.
+func (cs *ComplianceSuite) TestOnFinalizedBlock() {
+	finalizedBlock := unittest.BlockHeaderFixture()
+	cs.head = &finalizedBlock
+
+	done := atomic.NewBool(false)
+	cs.voteAggregator.On("PruneUpToView", finalizedBlock.View).Run(func(mock.Arguments) {
+		done.Toggle()
+	}).Return(nil).Once()
+	cs.engine.OnFinalizedBlock(finalizedBlock.ID())
+
+	// matching engine has at least 100ms ticks for processing events
+	time.Sleep(1 * time.Second)
+
+	require.Eventually(cs.T(), done.Load, time.Second, time.Millisecond*20)
```

Comment:
Same comment here - I think we could remove the sleep. 

```suggestion

	require.Eventually(cs.T(), done.Load, time.Second, time.Millisecond*20)
```